src/tests/tracing/eventpipe/allocationsampling/allocationsampling.cs

```diff
@@ -0,0 +1,255 @@
+            clock.Start();
+            int allocatedSize = AllocateObjects(InstanceCount * 50);
+            clock.Stop();
+            var duration = clock.ElapsedMilliseconds;
+            Logger.logger.Log($"       #GCs: {GC.CollectionCount(0)}");
+            Logger.logger.Log($"No Sampling: {InstanceCount * 50} instances allocated for {allocatedSize} bytes in {duration} ms");
+            _noSamplingDuration = duration;
+            Logger.logger.Log("-");
+
+            //// trigger GC to avoid one during the next tests
+            //GC.Collect();
+
+            var ret = IpcTraceTest.RunAndValidateEventCounts(
+                new Dictionary<string, ExpectedEventCount>() { { "Microsoft-Windows-DotNETRuntime", -1 } },
+                _eventGeneratingActionForAllocationsWithAllocationTick,
+                // GCKeyword (0x1): 0b1
+                new List<EventPipeProvider>() { new EventPipeProvider("Microsoft-Windows-DotNETRuntime", EventLevel.Verbose, 0b1) },
+                1024, _DoesTraceContainAllocationTickEvents, enableRundownProvider: false);
+            if (ret != 100)
+                return ret;
+            Logger.logger.Log("-");
+
+            //// trigger GC to avoid one during the next tests
+            //GC.Collect();
+
+            ret = IpcTraceTest.RunAndValidateEventCounts(
+                new Dictionary<string, ExpectedEventCount>() { { "Microsoft-Windows-DotNETRuntime", -1 } },
+                _eventGeneratingActionForAllocationsWithAllocationSampled,
+                // AllocationSamplingKeyword (0x80000000000): 0b1000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000
+                new List<EventPipeProvider>() { new EventPipeProvider("Microsoft-Windows-DotNETRuntime", EventLevel.Informational, 0x80000000000) },
+                1024, _DoesTraceContainAllocationSampledEvents, enableRundownProvider: false);
+            if (ret != 100)
+                return ret;
+
+            return 100;
+        }
+
+        private static int AllocateObjects(int instanceCount)
+        {
+            int size = 0;
+            for (int i = 0; i < instanceCount; i++)
+            {
+                byte[] bytes = new byte[35];
+                size += bytes.Length;
+            }
+
+            return size;
+        }
+
+        // 2000 instances of 1KB byte arrays should trigger around 20 events with the default 1/100KB sampling rate
+        const int InstanceCount = 2000;
+        const int MinExpectedEvents = 15;
+
+        // allocate objects to trigger dynamic allocation sampling events
+        private static Action _eventGeneratingActionForAllocations = () =>
+        {
+            int allocatedSize = 0;
+            for (int i = 0; i < InstanceCount; i++)
+            {
+                if ((i != 0) && (i % InstanceCount/5 == 0))
+                    Logger.logger.Log($"Allocated {i* InstanceCount / 5} instances {i} times...");
+
+                byte[] bytes = new byte[1024];
```

Comment:
I don't think the JIT optimized the allocations away because:
- the size of each allocated array is used to increment the count (I did that on purpose)
- and... I can see the events with the right type and size (and with a count that could not be explained by other allocations in the runtime) :^)
However, it is a good idea to allocate dedicated types with different sizes and orders of allocation (I did that to test the AllocationTick distribution - https://github.com/DataDog/dd-trace-dotnet/blob/master/profiler/src/Demos/Samples.Computer01/MeasureAllocations.cs). 
I will add such a test in a forthcoming commit

Reply:
The difference between our 2 different results maybe due to the fact that I'm using `set DOTNET_TieredCompilation=0` (otherwise I'm getting an error at the start of the test).
Anyway, I'm changing the code.